Client/Frontend/Browser/BrowserViewController.swift

```diff
@@ -531,6 +528,12 @@ class BrowserViewController: UIViewController {
                 profile.prefs.setString(appVersion, forKey: "latestAppVersion")
             }
         }
+
+        if var queuedAlertInfo = tabManager.selectedTab?.dequeueJavascriptAlertPrompt() {
+            let alertController = queuedAlertInfo.alertController()
+            alertController.delegate = self
```

Comment:
The delegate only gets fired on the dismissal of a prompt. In the case of switching to a new tab we need to check to see if there is anything to dequeue. I can probably extract this out into a small helper function.